From de90041ef3a5205fabb6dc61d017bbec95e3104a Mon Sep 17 00:00:00 2001
From: Ran Shalit <rshalit@habana.ai>
Date: Thu, 19 Nov 2020 17:58:03 +0200
Subject: [PATCH] [SW-14663] boot_desc: uncomment checksum

Uncomment memcpy for checksum verification,
so that checksum verification is fully done.
Previously we comment out the code because of
alignment issue. In order to enable it again,
we first solved the alignment issue in another
patch, and here we complete the enabling by
un-commenting memcpy.

Change-Id: I6dcd0db3b45eca647f8e013bd9ebe67373dc0a4a
Signed-off-by: Ran Shalit <rshalit@habana.ai>
---

diff --git a/common/boot_desc.c b/common/boot_desc.c
index 1131480..e8550d1 100644
--- a/common/boot_desc.c
+++ b/common/boot_desc.c
@@ -105,13 +105,8 @@
 		desc->loadaddr = ((u64)img_desc[i].load_high[idx] << 32) |
 					img_desc[i].load_low[idx];
 		desc->desc_ver = img_desc[i].desc_ver;
-
-		/*
-		 * Disable chksum copy temporarily, until alignment issue
-		 * is resolved.
-		 */
-		/* memcpy(desc->chksum, img_desc[i].chksum[idx], */
-		/* ARRAY_SIZE(desc->chksum)); */
+		memcpy(desc->chksum, img_desc[i].chksum[idx],
+		       ARRAY_SIZE(desc->chksum));
 
 		return idx;
 	}
